ledger/eval.go

```diff
@@ -733,7 +733,7 @@ func (eval *BlockEvaluator) transactionGroup(txgroup []transactions.SignedTxnWit
 		txibs = append(txibs, txib)
 
 		if eval.validate {
-			groupTxBytes += len(protocol.Encode(&txib))
+			groupTxBytes += txib.GetEncodedLength()
```

Comment:
extra brownie points ( overall solution ): we've currently triple-encoding the payset -
1. we encode it to calculate the block size.
2. we encode it to calculate the commit hash ( either via flat, or as Merkle tree ).
3. we encode it as a whole for the purpose of preparing the proposal ( I know that this isn't always the case, but this case happens to be on the critical path ).

Caching the encoded data of the *first* block could help us repeating the process.